src/OrchardCore/OrchardCore.Sms.Core/Services/TwilioSmsProvider.cs

```diff
@@ -79,17 +84,25 @@ public async Task<SmsResult> SendAsync(SmsMessage message)
         }
     }
 
-    private async Task EnsureClientIsIsInitializedAsync()
+    private async Task<TwilioSettings> GetSettingsAsync()
     {
-        if (_plainAuthToken == null)
+        if (_twilioSettings == null)
         {
             var settings = (await _siteService.GetSiteSettingsAsync()).As<TwilioSettings>();
 
-            _phoneNumber = settings.PhoneNumber;
-            var protector = _dataProtectionProvider.CreateProtector(SmsConstants.TwilioServiceName);
-            _plainAuthToken = protector.Unprotect(settings.AuthToken);
+            var protector = _dataProtectionProvider.CreateProtector(ProtectorName);
+
+            // It is important here to create a new instance of `TwilioSettings` privetly to hold the plain token value.
+            _twilioSettings = new TwilioSettings()
```

Comment:
`privetly` => `privately`

Better here to first init a variable `settings`, then uses it for the initialization and just after do `_twilioSettings = settings`, or maybe just `_settings = settings` as we are in the twilio provider.

Otherwise a thread may see `_twilioSettings` not null before it is initialized.

will review the rest asap.